src/aztec3/circuits/kernel/private/c_bind.test.cpp

```diff
@@ -0,0 +1,111 @@
+#include "c_bind.h"
+
+#include <aztec3/circuits/abis/call_context.hpp>
+#include <aztec3/circuits/abis/contract_deployment_data.hpp>
+#include <aztec3/circuits/abis/function_signature.hpp>
+#include <aztec3/circuits/abis/tx_context.hpp>
+#include <aztec3/circuits/abis/tx_request.hpp>
+#include <aztec3/circuits/abis/private_circuit_public_inputs.hpp>
+#include <aztec3/circuits/abis/private_kernel/private_inputs.hpp>
+#include <aztec3/circuits/abis/private_kernel/public_inputs.hpp>
+
+#include <aztec3/circuits/apps/function_execution_context.hpp>
+#include <aztec3/circuits/apps/test_apps/escrow/deposit.hpp>
+
+#include <common/test.hpp>
+#include <gtest/gtest.h>
+
+namespace {
+
+using aztec3::circuits::abis::CallContext;
+using aztec3::circuits::abis::ContractDeploymentData;
+using aztec3::circuits::abis::FunctionSignature;
+using aztec3::circuits::abis::OptionalPrivateCircuitPublicInputs;
+using aztec3::circuits::abis::PrivateCircuitPublicInputs;
+using aztec3::circuits::abis::TxContext;
+using aztec3::circuits::abis::TxRequest;
+
+using aztec3::circuits::apps::FunctionExecutionContext;
+using aztec3::circuits::apps::test_apps::escrow::deposit;
+
+} // namespace
+
+namespace aztec3::circuits::kernel::private_kernel {
+
+TEST(private_kernel, hash_tx_request)
+{
+    // Most of this is just setup (creating TxRequest to then be hashed)
+    const NT::address escrow_contract_address = 12345;
+
+    const NT::fr msg_sender_private_key = 123456789;
```

Comment:
I think we can make this randomised instead of a hard-coded value (just to make the test robust)

Reply:
Did this for all constant inputs (except 0/bool values): https://github.com/AztecProtocol/aztec3-circuits/pull/26/files#diff-14f002b1c41e9ac3c4fbb7a0991d2325b91f189442919413303740f6384e3e5fR43